core/silkworm/chain/config.hpp

```diff
@@ -19,164 +19,129 @@
+        "constantinopleBlock",  // EVMC_CONSTANTINOPLE
+        "petersburgBlock",      // EVMC_PETERSBURG
+        "istanbulBlock",        // EVMC_ISTANBUL
+        // there's no evmc_revision for muirGlacierBlock
+        "berlinBlock",  // EVMC_BERLIN
+    };
+
     // https://eips.ethereum.org/EIPS/eip-155
     uint64_t chain_id{0};
 
-    /*
-     * Note for developers !
-     * Adding/removing members here require to integrate Json() method
-     * in config.cpp *AND* in silkworm/db/common/access_layer.cpp
-     */
-
-    // https://eips.ethereum.org/EIPS/eip-606
-    std::optional<uint64_t> homestead_block;
-    bool has_homestead(uint64_t block_num) const noexcept {
-        return homestead_block.has_value() && homestead_block.value() <= block_num;
-    }
-
-    // https://eips.ethereum.org/EIPS/eip-608
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1015
-    std::optional<uint64_t> tangerine_whistle_block;
-    bool has_tangerine_whistle(uint64_t block_num) const noexcept {
-        return tangerine_whistle_block.has_value() && tangerine_whistle_block.value() <= block_num;
-    }
-
-    // TODO[ETC] EIP-160 was applied to ETC before the rest of Spurious Dragon; see
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1066
-
-    // https://eips.ethereum.org/EIPS/eip-607
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1054
-    std::optional<uint64_t> spurious_dragon_block;
-    bool has_spurious_dragon(uint64_t block_num) const noexcept {
-        return spurious_dragon_block.has_value() && spurious_dragon_block.value() <= block_num;
-    }
-
-    // https://eips.ethereum.org/EIPS/eip-609
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1054
-    std::optional<uint64_t> byzantium_block;
-    bool has_byzantium(uint64_t block_num) const noexcept {
-        return byzantium_block.has_value() && byzantium_block.value() <= block_num;
-    }
-
-    // https://eips.ethereum.org/EIPS/eip-1013
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1056
-    std::optional<uint64_t> constantinople_block;
-    bool has_constantinople(uint64_t block_num) const noexcept {
-        return constantinople_block.has_value() && constantinople_block.value() <= block_num;
-    }
-
-    // https://eips.ethereum.org/EIPS/eip-1716
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1056
-    std::optional<uint64_t> petersburg_block;
-    bool has_petersburg(uint64_t block_num) const noexcept {
-        return petersburg_block.has_value() && petersburg_block.value() <= block_num;
-    }
-
-    // https://eips.ethereum.org/EIPS/eip-1679
-    // https://ecips.ethereumclassic.org/ECIPs/ecip-1088
-    std::optional<uint64_t> istanbul_block;
-    bool has_istanbul(uint64_t block_num) const noexcept {
-        return istanbul_block.has_value() && istanbul_block.value() <= block_num;
-    }
+    // Block numbers of forks that have an evmc_revision value
+    std::array<std::optional<uint64_t>, EVMC_MAX_REVISION> fork_blocks{};
```

Comment:
Consider the addition of a method like this:
```
    std::optional<uint64_t> block_number(evmc_revision rev) const noexcept {
        return fork_blocks[rev];
    }
```
Although it is not necessary it better clarifies the link between fork_blocks array index and evmc_revision because in the following line it is not self evident (only the constant EVMC_MAX_REVISION suggests it):

`std::array<std::optional<uint64_t>, EVMC_MAX_REVISION> fork_blocks{};`

It would also be nice in all software to be able to use

`using BlockNum = uint64_t;`

so as to distinguish a uint64_t which is a block number from one which is something else. 
But this is a change that I would leave for the future

Reply:
@mriccobene is this what you had n mind ?
https://github.com/torquem-ch/silkworm/pull/227/commits/68532661326c51decf8950724a8cd0195f9cb420

Reply:
@AndreaLanfranchi yes! Now that I see my bug I think this method is necessary! Thanks!